### PR TITLE
[Py OV] Add op to openvino module

### PR DESCRIPTION
**Pull request by almilosz**
### Details:
- copy `openvino/runtime/op/` to `openvino`. `op/` is not initialized in `openvino/runtime/__init__.py`. Update inits in openvino/runtime/op/
- update rule `src/openvino/runtime/*/ops.py: VNE001,VNE003` in _setup.cfg_
- For opset 1 - 16:
  - move `runtime/opsetx/ops.py`. 
  - update opsetx inits to import from new ops.py location. 
  - add `runtime/opsetx/ops/__init__.py`.
- initialize opsets in openvino init
- add imports for opset14-16 to `openvino/runtime/__init__.py`
- adds possibility to import opsets directly:
```python
import openvino.opset13 as ops
param = ops.parameter([10])
```
or
```python
import openvino as ov
param = ov.opset13.parameter([10])
```

### Tickets:
 - CVS-129458

### CODE REVIEW

File: src/bindings/python/src/openvino/op/__init__.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Package: openvino.op
+Low level wrappers for the c++ api in ov::op.
+"""
+
+# flake8: noqa
+
+from openvino._pyopenvino.op import Constant
```
Comment (p-wysocki) [inline src/bindings/python/src/openvino/op/__init__.py:11]: Can these be imported directly like in `src/bindings/python/src/openvino/runtime/op/__init__.py`?
`from openvino.op import Constant`

File: src/bindings/python/src/openvino/__init__.py [hunk 2]
```diff
@@ -56,6 +56,24 @@
...
+from openvino import opset7
+from openvino import opset8
+from openvino import opset9
+from openvino import opset10
+from openvino import opset11
+from openvino import opset12
+from openvino import opset13
+from openvino import opset14
+from openvino import opset15
+from openvino import opset16
```
Comment (mitruska) [inline src/bindings/python/src/openvino/__init__.py:75]: Could you please explain why those (all opsets) imports are needed here (and in `benchmark_tool/openvino/__init__.py`,  `ovc/openvino/__init__.py`)?

**Review by mitruska — COMMENTED** [review]
In the ticket description the desired import statement is `openvino.opsets.opsetX` (which is kind of aligned with [C++ headers](
https://github.com/openvinotoolkit/openvino/blob/9559b421950c09d44b614958cec5f7ec631cadf7/src/plugins/intel_cpu/src/shape_inference/shape_inference.cpp#L9-L11)), but the PR makes it `openvino.opsetX` (without the mid `opsets`).
Is that intended?

The parent "opsets" directory for the opsetX dirs,  could make the main "openvino" dir less messy.

File: src/bindings/python/src/openvino/runtime/opset1/__init__.py
[the remark this answers is not in the document]
Comment (akuporos) [reply src/bindings/python/src/openvino/runtime/opset1/__init__.py:1]: Deprecation warning will be added in a separate PR

File: src/bindings/python/src/openvino/__init__.py (hunk 2 above)
Comment (akuporos) [reply src/bindings/python/src/openvino/__init__.py:75]: It's needed if someone wants to use opsets like this:
```python
>>> import openvino as ov
>>> ov.opset15.constant([1,2,3])
```

Comment (mlukasze) [reply src/bindings/python/src/openvino/__init__.py:75]: there is no other way to import those opsets collections dynamically, on demand?

File: src/bindings/python/src/openvino/op/__init__.py (hunk 1 above)
Comment (almilosz) [reply src/bindings/python/src/openvino/op/__init__.py:11]: I believe here we want to import "base" `Op` directly from `_pyopenvino` while in `openvino/runtime/op/__init__.py` we want to have the same `Op` as in `openvino/op/__init__.py`

File: src/bindings/python/src/openvino/__init__.py (hunk 2 above)
Comment (almilosz) [reply src/bindings/python/src/openvino/__init__.py:75]: Init files for `openvino`, `benchmark_tool.openvino` and `ovc/openvino` have to be the same and it's enforced by cmake rule.

File: src/bindings/python/src/openvino/runtime/opset1/__init__.py [hunk 3]
```diff
@@ -2,110 +2,110 @@
...
-from openvino.runtime.opset1.ops import tile
-from openvino.runtime.opset1.ops import topk
-from openvino.runtime.opset1.ops import transpose
-from openvino.runtime.opset1.ops import unsqueeze
-from openvino.runtime.opset1.ops import variadic_split
+from openvino.opset1.ops import absolute
+from openvino.opset1.ops import absolute as abs
+from openvino.opset1.ops import acos
+from openvino.opset1.ops import add
+from openvino.opset1.ops import asin
```
[the remark this answers is not in the document]
Comment (almilosz) [reply src/bindings/python/src/openvino/runtime/opset1/__init__.py:9]: I think that we won't add new functionality to deprecated `openvino.runtime` so there won't be 32 files to manage

[the remark this answers is not in the document]
Comment (akuporos) [reply src/bindings/python/src/openvino/runtime/opset1/__init__.py:9]: As it also was discussed in the original epic, the `openvino.runtime` submodule will be freezed

File: src/bindings/python/src/openvino/__init__.py (hunk 2 above)
Comment (mitruska) [reply src/bindings/python/src/openvino/__init__.py:75]: Assuming that model inference is the main use case of openvino, I believe there is no need to import all of the opsets in openvino `__init__.py` file. A single opsetX can be imported by user as needed, without affecting the import time of the main openvino module (especially that the number of the opsets can grow).

File: src/bindings/python/src/openvino/opset1/ops.py [hunk 4]
```diff
@@ -9,7 +9,7 @@
 from functools import partial
 
 from openvino.runtime import Node, PartialShape, Type
-from openvino.runtime.op import Constant, Parameter, tensor_iterator
+from openvino.op import Constant, Parameter, tensor_iterator
```
Comment (ilya-lavrenov) [inline src/bindings/python/src/openvino/opset1/ops.py:12]: why Ops like Parameter are located w/o version namespace ? In C++ it's v0

If we introduce v20.Parameter, how to understand what parameter is used in this file?

Comment (almilosz) [reply src/bindings/python/src/openvino/opset1/ops.py:12]: Let's discuss it in a new task. Here I want to move `opsets` to new location without introducing new logic

**Review by akuporos — APPROVED** [review]

**Comment by slyalin** [comment]
> > In the ticket description the desired import statement is `openvino.opsets.opsetX` (which is kind of aligned with [C++ headers](https://github.com/openvinotoolkit/openvino/blob/9559b421950c09d44b614958cec5f7ec631cadf7/src/plugins/intel_cpu/src/shape_inference/shape_inference.cpp#L9-L11)), but the PR makes it `openvino.opsetX` (without the mid `opsets`). Is that intended?
> 
> Yes. In the epic description there is a [comment](https://jira.devtools.intel.com/browse/CVS-129450?focusedId=22697983&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-22697983) to skip mid `opsets` and it's aligned with migration rules. cc: @ilya-lavrenov
> 
> > The parent "opsets" directory for the opsetX dirs, could make the main "openvino" dir less messy.
> 
> Here I think we should focus how the opsets will be accessible and if opsetx should be initialized alongside openvino (even if the user does not use it).
> 
> Structure	`opsetx` submodules initialized in `src/bindings/python/src/openvino/__init__.py`	Not Initialized
> Mid `opsets` directory	`import openvino as ov`
> `ov.opset15.constant([1,2,3])`
> and
> `import openvino.opset15`
> plus
> `import openvino.opsets.opset15`	`import openvino.opsets.opset15`
> Flat structure	`import openvino.opset15`
> and
> `import openvino as ov`
> `ov.opset15.constant([1,2,3])`	`import openvino.opset15`

In favor of `openvino.opset15` syntax.

**Review by mitruska — APPROVED** [review]
My opinion is as in the comment https://github.com/openvinotoolkit/openvino/pull/27902#discussion_r1875897536.

Approved, as the difference in the import time has been identified as not significant, and skipping direct import of `openvino.opsetX.theop` can be preferred by some users.

**Review by slyalin — APPROVED** [review]
I am voting for the default availability of syntax `ov.opsetX.gather` with just a single `import openvino as ov`. So don't want to have an extra explicit import for `opsetX`.

`ov.opsetX` is more preferable than `ov.opsets.opsetX`, don't see clear purpose behind `ov.opset`.

For future improvements I would suggest to think about more clearer availability of operations without mentioning opsets but having versions/opsets easily plugged in if they are required for each op individually. My personal preference would be to have something like `ov.ops.gather.v8(args)` or even `ov.ops.gather.opset15(args)` when I want to mention a specific version of operation, and `ov.ops.gather(args)` when I want to use the latest one. The form `ov.ops.gather.version` gives you a clear way to list all different available versions of the op in the autocompletion in your code editor when you start typing `ov.ops.gather.`, without it you need to use doc/ov source each time. This approach could be propagated back to C++ but it requires a redesign of how you are instantiating OV operations in C++ code.

**Comment by mryzhov** [comment]
Those changes broke genai pipeline:
ImportError: cannot import name '_get_node_factory_opset1' from 'openvino.runtime.opset1.ops' (/venv/lib/python3.11/site-packages/openvino/runtime/opset1/ops/__init__.py)

**Comment by mlukasze** [comment]
fix is on the way, sorry for that

**Pull request merged**